layers/ports.go

```diff
@@ -61,20 +61,20 @@ func (a UDPPort) String() string {
 //
 // Returns gopacket.LayerTypePayload for unknown/unsupported port numbers.
 func (a UDPPort) LayerType() gopacket.LayerType {
-	switch a {
-	case 53:
-		return LayerTypeDNS
-	case 123:
-		return LayerTypeNTP
-	case 4789:
-		return LayerTypeVXLAN
-	case 67, 68:
-		return LayerTypeDHCPv4
-	case 6343:
-		return LayerTypeSFlow
-	default:
-		return gopacket.LayerTypePayload
+	lt := udpPortLayerType[uint16(a)]
+	if lt != 0 {
+		return lt
 	}
+	return gopacket.LayerTypePayload
+}
+
+var udpPortLayerType [65536]gopacket.LayerType
+
+// RegisterUDPPortLayerType create a new mapping between an UDPPort
+// and an underlaying LayerType.
+func RegisterUDPPortLayerType(portNum uint16, layerType gopacket.LayerType) uint16 {
```

Comment:
Please accept a layers.UDPPort instead of a uint16 here.

I'd also maybe remove the return value... doesn't seem too necessary/useful.